src/guardian/keys-validation/keys-validation.service.ts

```diff
@@ -62,4 +112,53 @@ export class KeysValidationService {
 
     return forkVersion;
   }
+
+  async updateCacheWithValidationResults(
+    validatedKeys: [Key & DepositData, boolean][],
+  ) {
+    validatedKeys.forEach(([key, isValid]) =>
+      this.keysCache.set(key.key, { signature: key.depositSignature, isValid }),
+    );
+  }
+
+  private mergeInvalidKeys(
+    vettedKeys: RegistryKey[],
+    recentInvalidKeys: { key: string; depositSignature: string }[],
+  ): { key: string; depositSignature: string }[] {
```

Comment:
1. Excude checked keys (in cache)
2. Validate keys
3. Write to cache
4. Cachec is overfilled. Lastest keys are deleted from cache.
5.  There were some invalid keys
6. Council daemon send message even it shouldn't

Reply:
I propose to rewrite and do next things
1. Get results from cache filtered by vetted keys
2. Filter not validated keys and validated them
3. merge two lists

Reply:
agree, here could be a problem